discord-scripts/fjord.ts

```diff
@@ -0,0 +1,334 @@
+
+    const handleThreadCreate = async (
+      message: any,
+      threadName: any,
+      threadReason: any,
+      responseData: any,
+    ) => {
+      const debugThread = await message.startThread({
+        name: threadName,
+        autoArchiveDuration: 60,
+        reason: threadReason,
+      })
+      await debugThread.send("Over here!")
+      await debugThread.send(responseData)
+    }
+
+    if (message.content === "!ping") {
+      channel.send({
+        content: "ping pong ping ping pong ping / o  o",
+        components: [
+          {
+            type: 1,
+            components: [
+              {
+                type: 2,
+                label: "Ping?",
+                style: 1,
+                custom_id: "ping",
+              },
+            ],
+          },
+        ],
+      })
+    }
+
+    if (message.content === "!debug") {
+      console.log("adapter in use:", robot.adapter)
+      channel.send({
+        content: "**Debugger running, check your console ;)**",
+        components: [
+          {
+            type: 1,
+            components: [
+              {
+                type: 2,
+                label: "Run debugger again",
+                style: 1,
+                custom_id: "debug",
+              },
+            ],
+          },
+        ],
+      })
+    }
+
+    if (message.content === "!fjord") {
+      channel.send("**Fjord is listening, what do you want?**")
+      channel.send(
+        "Here's a list of Fjord commands. !issues <repo-owner> <repo-name> !stale-issues <repo-owner> <repo-name> !debug !activity <repo-owner> <repo-name>",
+      )
+    }
+
+    if (message.content === "!thread") {
+      if (message.channel.type === ChannelType.GuildPublicThread) {
+        channel.send("You can't thread a thread!")
+      }
+      const threadName = "New thread"
+      await handleThreadCreate(
+        message,
+        threadName,
+        "thread test",
+        "Setup a thread for you",
+      )
+    }
+
+    if (message.content.startsWith("!stale-issues")) {
+      const parameters = message.content.split(" ").slice(1)
+      if (parameters.length >= 2) {
+        const [repositoryOwner, repositoryName] = parameters
+        const webhookUrl =
+          "http://n8n.thesis.co/webhook/ec766fde-4ce5-4679-8a50-462e9e68e16a"
+        const queryParams = new URLSearchParams({
+          repositoryOwner: repositoryOwner,
+          repositoryName: repositoryName,
+        })
+
+        message.reply({
+          content: `**Stale issue automation:  ${repositoryOwner} / ${repositoryName}**`,
+          components: [
+            {
+              type: 1,
+              components: [
+                {
+                  type: 2,
+                  label: "Run again",
+                  style: 1,
+                  custom_id:
+                    "stale-issues " + repositoryOwner + " " + repositoryName,
+                },
+              ],
+            },
+          ],
+        })
+        axios
```

Comment:
not to be overly nit-picky but I'd love to nail down the use of fetch vs axios to one type we use, I tend towards a wrapping w/ fetch but i'm not dogmatic about it

Reply:
@puppycodes Thanks for all the great feedback! Just deployed changes to latest commit. Happy to switch over to `fetch()` but was wondering is `axios` going to be better for us to handle timeout events or does it matter?

Reply:
I'm open to either! I know its one less dependency which attracts me to fetch but I feel like its more a pref. It's nbd just curious if we should choose one. I'd say fetch would be my pref because it removes a dependency and use this technique https://davidwalsh.name/fetch-with-timeout

Reply:
Did some testing as was wondering why `fetch()` isn't working out of the box. Seems like it's actually a ESM conflict and valkyrie instance has `ts-node/esm --no-experimental-fetch` flags which disables built in `fetch`. Tested without and it seems to break / create conflicts with hubot typescript modules. In that case might make sense to stick with `axios`?